common/src/View/RenderView.cpp

```diff
@@ -121,37 +121,46 @@ namespace TrenchBroom {
         RenderView::~RenderView() = default;
 
         void RenderView::keyPressEvent(QKeyEvent* event) {
-            m_eventRecorder.recordEvent(event);
+            m_eventRecorder.recordEvent(*event);
             update();
         }
 
         void RenderView::keyReleaseEvent(QKeyEvent* event) {
-            m_eventRecorder.recordEvent(event);
+            m_eventRecorder.recordEvent(*event);
             update();
         }
 
+        static auto mouseEventWithFullPrecisionLocalPos(const QWidget* widget, const QMouseEvent* event) {
+            // The localPos of a Qt mouse event is only in integer coordinates, but window pos
+            // and screen pos have full precision. We can't directly map the windowPos because
+            // mapTo takes QPoint, so we just map the origin and subtract that.
+            const auto localPos = event->windowPos() - QPointF(widget->mapTo(widget->window(), QPoint(0, 0)));
+            std::cout << "window pos: " << event->windowPos().x() << "," << event->windowPos().y() << " - local pos: " << localPos.x() << "," << localPos.y() << std::endl;
```

Comment:
I'd suggest `#if 0`  (or other ifdef) around this logging, rather than deleting, as it's annoying to have write it again.

Reply:
I left that by accident. I prefer not to leave these in the code. It takes a minute to write, but it's not likely we'll need it very often.